x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/HttpCertificateCommand.java

```diff
@@ -0,0 +1,1149 @@
+        terminal.println("sign your new http certificate. This allows you to use the same CA across");
+        terminal.println("multiple Elasticsearch clusters which can make it easier to configure clients,");
+        terminal.println("and may be easier for you to manage.");
+        terminal.println("");
+        terminal.println("If you do not have an existing CA, one will be generated for you.");
+        terminal.println("");
+
+        return terminal.promptYesNo("Use an existing CA?", false);
+    }
+
+    private <T> T tryReadInput(Terminal terminal, String prompt, T defaultValue, Function<String, T> parser) {
+        final String defaultStr = defaultValue instanceof Period ? toString((Period) defaultValue) : String.valueOf(defaultValue);
+        while (true) {
+            final String input = terminal.readText(prompt + " [" + defaultStr + "] ");
+            if (Strings.isEmpty(input)) {
+                return defaultValue;
+            }
+            T parsed = parser.apply(input);
+            if (parsed != null) {
+                return parsed;
+            }
+        }
+    }
+
+    static String toString(Period period) {
+        if (period == null) {
+            return "N/A";
+        }
+        if (period.isZero()) {
+            return "0d";
+        }
+        List<String> parts = new ArrayList<>(3);
+        if (period.getYears() != 0) {
+            parts.add(period.getYears() + "y");
+        }
+        if (period.getMonths() != 0) {
+            parts.add(period.getMonths() + "m");
+        }
+        if (period.getDays() != 0) {
+            parts.add(period.getDays() + "d");
+        }
+        return Strings.collectionToCommaDelimitedString(parts);
+    }
+
+    private Path requestPath(String prompt, Terminal terminal, Environment env, boolean requireExisting) {
+        for (; ; ) {
+            final String input = terminal.readText(prompt);
+            final Path path = env.configFile().resolve(input).toAbsolutePath();
+
+            if (requireExisting == false || Files.isReadable(path)) {
+                return path;
+            }
+
+            if (Files.notExists(path)) {
+                terminal.println(Terminal.Verbosity.SILENT, "The file " + path + " does not exist");
+            } else {
+                terminal.println(Terminal.Verbosity.SILENT, "The file " + path + " cannot be read");
+            }
+        }
+    }
+
+    static FileType guessFileType(Path path, Terminal terminal) {
+        // trust the extension for some file-types rather then inspecting the contents
+        // we don't rely on filename for PEM files because
+        // (a) users have a tendency to get things mixed up (e.g. naming something "key.crt")
+        // (b) we need to distinguish between Certs & Keys, so a ".pem" file is ambiguous
+        final String fileName = path == null ? "" : path.getFileName().toString().toLowerCase(Locale.ROOT);
```

Comment:
`path.getFileName()` can return `null` ( if someone enters a path without a file name ). While such an input is wrong, we could handle it an not fail with an NPE

Reply:
I'll put a fix in `requestPath` to make sure the path always has a file part.